src/components/Channel/Channel.tsx

```diff
@@ -45,25 +224,46 @@ const Channel = ({ EmptyPlaceholder = null, ...props }) => {
   return <ChannelInner {...props} channel={channel} key={channel.cid} />;
 };
 
-/** @type {React.FC<import('types').ChannelProps & { channel: import('stream-chat').Channel }>} */
-const ChannelInner = ({
-  Attachment = DefaultAttachment,
-  doMarkReadRequest,
-  Emoji = DefaultEmoji,
-  emojiData = defaultEmojiData,
-  EmojiIndex = DefaultEmojiIndex,
-  EmojiPicker = DefaultEmojiPicker,
-  LoadingErrorIndicator = DefaultLoadingErrorIndicator,
-  LoadingIndicator = DefaultLoadingIndicator,
-  Message = MessageSimple,
-  ...props
-}) => {
-  const { channel } = props;
-
-  const { client, mutes, theme } = useContext(ChatContext);
-  const { t } = useContext(TranslationContext);
-
-  const [state, dispatch] = useReducer(channelReducer, initialState);
+const ChannelInner = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  props: PropsWithChildren<ChannelProps<At, Ch, Co, Ev, Me, Re, Us>> & {
+    key: string;
+  },
```

Comment:
not gonna be properly formatted with prettier but think we should do it this way
```suggestion
  props: PropsWithChildren<ChannelProps<At, Ch, Co, Ev, Me, Re, Us> & {
    key: string;
  }>,
```